File: src/main/java/PomadaWassington.java
public class PomadaWassington extends DecoratorAtaques {
    PomadaWassington(AtaqueConcret aDecorar) {
        super(aDecorar);
    }
    public void realizarAtaque(Chinpokomon atacado, Chinpokomon atacante) {
        var total = this.getRecupera();
        if(this.tieneVentaja(atacado,atacante)){
            total += this.getVentaja();
            atacante.modificarVida( this.getRecupera() + this.getVentaja() );
        }else{
            atacante.modificarVida( + this.getRecupera());
        }
        atacado.setRecuperacionRecibido( total );
    }
}

File: python/PomadaWassington.py
from DecoratorAtaques import DecoratorAtaques

class PomadaWassington(DecoratorAtaques):
    def __init__(self, aDecorar):
        super().__init__(aDecorar)
    
    def realizar_ataque(self, atacado, atacante) :
        total = self.get_recupera()
        if self.tiene_ventaja(atacado,atacante):
            total += self.get_ventaja()
            atacante.modificar_vida( total )            
        else:
            atacante.modificar_vida( total )
        atacado.set_recuperacion_recibido( total )